### ISSUE TITLE
Always add whitespace between the stoichiometric number and the compound

### ISSUE THREAD

**Issue by BackThePortal** [schang412/typst-whalogen#12]
The manual says that this is _a characteristic of Typst_, but given that this is a library specific for chemical formulas, this behaviour should be fixed. I'm sure some trick can be done to fix this.

**Comment by BackThePortal** [comment]
The solution is to always use `" "` instead of just `" "` when a whitespace is used. I would do it myself and make a PR, but I don't understand the code, lol.

**Comment by BackThePortal** [comment]
Replacing `_buffer = ""`  `_buffer = _quote(" ")` adds this whitespace, but I don't know how to remove the conventional one.
```typst
// end previous states on whitespace
  if _char_in.contains(regex("\s")) {
    _out = _flush_ce_buffer(_state, _buffer)
    _buffer = _quote(" ")
    _state = ""
  }
```

So I get this:
![image](https://github.com/user-attachments/assets/3e9072cb-26ae-4c65-9bc1-ef29da1c3552)

Where is this whitespace added?

**Comment by schang412** [comment]
I’ll look into this and see if I can fix it, but I will be fairly busy for at least the next 2 weeks. I believe I said it’s a characteristic of Typst because `2^2(3)` will move everything up while `2^2 (3)` will split the two.

**Comment by BackThePortal** [comment]
> I’ll look into this and see if I can fix it, but I will be fairly busy for at least the next 2 weeks. I believe I said it’s a characteristic of Typst because `2^2(3)` will move everything up while `2^2 (3)` will split the two.

Okay, no worries.

I think always using `" "` works and looks fine. I've never written any Typst code like this, but if I manage to understand the code, I will make a PR.

**Comment by schang412** [comment]
Could you expand on what you are trying to accomplish here? Perhaps it would be easiest to show two compounds which have the behavior you want to see fixed. Thanks.

**Comment by BackThePortal** [comment]
> Could you expand on what you are trying to accomplish here? Perhaps it would be easiest to show two compounds which have the behavior you want to see fixed. Thanks.

Sure!

![image](https://github.com/user-attachments/assets/2209c927-f6c0-4c48-9dd3-d122ff0c0e30)

My suggestion is using a quoted space `" "` between the stoichiometric number and the compound instead of a normal space.

**Comment by svanterwingen** [comment]
I agree, there should be a space between the sum formula and stoichiometric coefficient. Maybe even a non-breaking half space, as implemented in mhchem (I guess):

<img width="840" height="75" alt="Image" src="https://github.com/user-attachments/assets/b36b8e98-bce1-4ad1-a713-2cbbb65782f2" />
